Porting auto backup oca server-tools

archetipo opened the pull request:
This PR  contains the porting form  [Yenthe666 /auto_backup](https://github.com/Yenthe666/auto_backup),
and it was preserved the history of commits.

need review

Yenthe666 committed:
Update english pot file

Added all the new fields and sentences. This will be the template for
translations.

Yenthe666 committed:
Update english pot file

Added all the new fields and sentences. This will be the template for
translations.

Yenthe666 committed:
Dutch translations

Full translation of the module to Dutch

Yenthe666 committed:
Dutch translations

Full translation of the module to Dutch

Yenthe666 committed:
Chinese translations

Add Chinese translations to the module. Written by talway.

Yenthe666 committed:
Chinese translations

Add Chinese translations to the module. Written by talway.

Yenthe666 committed:
Changes chinese translation

Changed some translations

Yenthe666 committed:
Changes chinese translation

Changed some translations

Yenthe666 committed:
Full German translations

Thanks to Martin Schmid!

Yenthe666 committed:
Full German translations

Thanks to Martin Schmid!

Yenthe666 committed:
Typo fix

could'nt > couldn't

Yenthe666 committed:
Typo fix

could'nt > couldn't

Yenthe666 committed:
Flemish translations

Flemish translations are identical to Dutch.

Yenthe666 committed:
Flemish translations

Flemish translations are identical to Dutch.

archetipo committed:
Open FTP session on the last moment possible

Used to open  fp = open(file_path,'wb') a few lines before it was needed. This shouldn't be too much of a problem but opening and closing it right after eachother keeps the session open for less time and there are less chances on failure.

archetipo committed:
Open FTP session on the last moment possible

Used to open  fp = open(file_path,'wb') a few lines before it was needed. This shouldn't be too much of a problem but opening and closing it right after eachother keeps the session open for less time and there are less chances on failure.

archetipo committed:
Porting module to OCA 8.0

archetipo committed:
Porting module to OCA 8.0

archetipo committed:
[FIX] module

archetipo committed:
[FIX] module

archetipo committed:
[FIX] bug logger --> _logger call
[FIX] Flake8
[RM] useless files

archetipo committed:
[FIX] bug logger --> _logger call
[FIX] Flake8
[RM] useless files

archetipo committed:
[FIX] Readme.rst

archetipo committed:
[FIX] Readme.rst

archetipo committed:
[FIX] defaults value

archetipo committed:
[FIX] defaults value

archetipo committed:
[FIX] rebased commit

archetipo committed:
[FIX] rebased commit

archetipo committed:
[RM] description index.html

archetipo committed:
[RM] description index.html

archetipo committed:
[RF] porting to new api
[FIX] schedule_backup method
[IMP] IT translation
[IMP] tests

archetipo committed:
[RF] porting to new api
[FIX] schedule_backup method
[IMP] IT translation
[IMP] tests

archetipo committed:
[FIX] test

archetipo committed:
[FIX] test

archetipo committed:
[FIX] flake8

archetipo committed:
[FIX] flake8

archetipo committed:
[IMP] deps in travis.yml
[FIX] flake8

archetipo committed:
[IMP] deps in travis.yml
[FIX] flake8

archetipo commented:
PR is ready and need review,tnx
TODO improve test case

archetipo committed:
[FIX] flake8 and pylint

archetipo committed:
[FIX] flake8 and pylint

eLBati commented on auto_backup/README.rst:
```diff
@@ -0,0 +1,99 @@
+Contributors
+------------
+
+* Yenthe Van Ginneken <email.address@example.org>
```
`email.address@example.org` should be removed or replaced by @Yenthe666 email

Yenthe666 replied:
My e-mail is yenthe.vanginneken@vanroey.be @eLBati :+1:

archetipo committed:
[FIX] name of file

archetipo committed:
[FIX] name of file

archetipo committed:
[FIX] autoremove method
[FIX] Contributors

archetipo committed:
[FIX] autoremove method
[FIX] Contributors

dreispt commented on auto_backup/README.rst:
```diff
@@ -0,0 +1,99 @@
+
+To configure this module, you need to:
+
+* go to ...
```
??

archetipo committed:
[FIX] mail.tempale seems not work in cron task, replaced  with direct call of mail.mail

archetipo committed:
[FIX] mail.tempale seems not work in cron task, replaced  with direct call of mail.mail

archetipo committed:
[FIX] Readme

archetipo committed:
[FIX] Readme

eLBati commented on auto_backup/i18n/it.po:
```diff
@@ -0,0 +1,363 @@
+"                 <p>Dettaglio errore: ${object.lasterrorlog}</p>\n"
+"                 <p>Cordiali Saluti</p>\n"
+"\n"
+"        "
```
It seems this string is not present in module, so it.po should be updated

archetipo replied:
@eLBati  thanks yes

Yenthe666 commented:
@eLBati I've added an improvement to prevent sockets from being closed. Some servers automatically do this and this line will keep it open.
https://github.com/Yenthe666/auto_backup/commit/9737681dde35d24ef5ead520a6dbfd76ee79bc70

gurneyalex commented on auto_backup/model/backup_scheduler.py:
```diff
@@ -0,0 +1,405 @@
+        'This module needs pysftp to automaticly write backups to the FTP '
+        'through SFTP.Please install pysftp on your system.'
+        '(sudo pip install pysftp)'
+    )
```
Raise an exception here is a :bug:  Odoo will check that pysftp is present when installing. However you need to catch ImportError to avoid crashing the instance if pysftp is not available and the addon is not installed either.

gurneyalex commented on auto_backup/model/backup_scheduler.py:
```diff
@@ -0,0 +1,405 @@
+    host = fields.Char(
+        string='Host', default='localhost', size=100, required=True)
+    port = fields.Char(
+        string='Port', default='8069', size=10, required=True)
```
I'm surprised these fields are needed at all.

archetipo replied:
@gurneyalex  thanks. why  are you surprised?

gurneyalex replied:
The port is available in the configuration of the server, and for host, localhost should always work, unless you explicitely disable listening on 127.0.0.1 (which is usually the only interface which is kept when securing a server behind a reverse proxy) -> I don't see why you need to configure this.

gurneyalex commented on auto_backup/model/backup_scheduler.py:
```diff
@@ -0,0 +1,405 @@
+            "should be made with. This is the password from the user"
+            " on the external server."
+        )
+    )
```
is public key authentication supported?

archetipo replied:
no, isn't.

yajo replied:
That would be awesome.

gurneyalex commented on auto_backup/model/backup_scheduler.py:
```diff
@@ -0,0 +1,405 @@
+    _constraints = [
+        (
+            _check_db_exist,
+            _('Error ! No such database exists!'), ['name']
```
please remove exclamation marks.

archetipo replied:
ok

gurneyalex commented on auto_backup/model/backup_scheduler.py:
```diff
@@ -0,0 +1,405 @@
+                    messageContent += _(
+                        "\nYour IP address seems to be too short.\n")
+                messageContent += "Here is what we got instead:\n"
+        if "Failed" in messageTitle:
```
does this work in another language than english?

gurneyalex commented on auto_backup/model/backup_scheduler.py:
```diff
@@ -0,0 +1,405 @@
+            "The IP address from your remote"
+            " server. For example 192.168.0.1"
+        )
+    )
```
not possible to work with a server name?

archetipo replied:
@gurneyalex  yes  why not?

gurneyalex replied:
well everything points to "IP address" not to "DNS name" so, stumbling on this while reading the code I'm wondering if there is something which prevents using a FQDN.

yajo replied:
I'd prefer the more generic "host" word too.

gurneyalex commented on auto_backup/model/backup_scheduler.py:
```diff
@@ -0,0 +1,405 @@
+                    if not os.path.isdir(rec.bkp_dir):
+                        os.makedirs(rec.bkp_dir)
+                except:
+                    raise
```
is this useful? Why don't you let the exception travel up?

gurneyalex commented on auto_backup/model/backup_scheduler.py:
```diff
@@ -0,0 +1,405 @@
+                bkp = ''
+                try:
+                    bkp = execute(
+                        conn, 'dump', tools.config['admin_passwd'], rec.name)
```
so you expect the remote admin password to be the same as the local admin password? this needs to be documented.

gurneyalex commented on auto_backup/model/backup_scheduler.py:
```diff
@@ -0,0 +1,405 @@
+                    bkp = execute(
+                        conn, 'dump', tools.config['admin_passwd'], rec.name)
+                except:
+                    _logger.notifyChannel(
```
deprecated I think

gurneyalex commented on auto_backup/model/backup_scheduler.py:
```diff
@@ -0,0 +1,405 @@
+                fp.write(bkp)
+                fp.close()
+            else:
+                _logger.notifyChannel(
```
deprecated.

gurneyalex commented on auto_backup/model/backup_scheduler.py:
```diff
@@ -0,0 +1,405 @@
+                    # Move to the correct directory on external server. If the
+                    # user made a typo in his path with multiple slashes
+                    # (/odoo//backups/) it will be fixed by this regex.
+                    pathToWriteTo = re.sub('([/]{2,5})+', '/', pathToWriteTo)
```
Are you sure you don't want just `re.sub('/+', '/', pathToWriteTo)`?

gurneyalex replied:
I'm pretty sure you do want this: your code handles 5 consecutive slashes, 7 consecutive slashes, but not 6 :smile_cat:

gurneyalex commented on auto_backup/model/backup_scheduler.py:
```diff
@@ -0,0 +1,405 @@
+                    # We will now check if the user wants
+                    # an e-mail notification about this.
+                    if rec.sendmailsftpfail:
+                        try:
```
extract this block in a separate method.

archetipo committed:
[FIX] review remarks

archetipo committed:
[FIX] review remarks

archetipo committed:
[FIX] handled ssl hosts
[FIX] Flake8

archetipo committed:
[FIX] handled ssl hosts
[FIX] Flake8

archetipo committed:
[FIX] handled ssl hosts
[FIX] Flake8

archetipo committed:
[FIX] handled ssl hosts
[FIX] Flake8

archetipo commented:
@gurneyalex @eLBati 
Thanks for reviews.
I've updated the code.

eLBati commented:
@archetipo `This branch has conflicts that must be resolved` , please rebase

archetipo commented:
thanks @eLBati  rebase was done

eLBati commented:
:+1:

tafaRU commented on auto_backup/README.rst:
```diff
@@ -0,0 +1,99 @@
+    :alt: License: AGPL-3
+
+Automated backups
+===========
```
Please fix the syntax according to https://raw.githubusercontent.com/OCA/maintainer-tools/master/template/module/README.rst

tafaRU commented on auto_backup/__init__.py:
```diff
@@ -0,0 +1,25 @@
+
+from . import model
+from . import tests
+# vim:expandtab:smartindent:tabstop=4:softtabstop=4:shiftwidth=4:
```
remove vim commands please

tafaRU commented on auto_backup/__openerp__.py:
```diff
@@ -0,0 +1,43 @@
+
+{
+    "name": "Database Auto-Backup",
+    "version": "1.0",
```
Please update it according to https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#version-numbers

tafaRU commented on auto_backup/__openerp__.py:
```diff
@@ -0,0 +1,43 @@
+    "category": "Tools",
+    "summary": "Backups data base",
+    "depends": ['base', 'email_template'],
+    "demo": [],
```
Please remove it

tafaRU commented on auto_backup/__openerp__.py:
```diff
@@ -0,0 +1,43 @@
+    "active": False,
+    "installable": True
+}
+# vim:expandtab:smartindent:tabstop=4:softtabstop=4:shiftwidth=4:
```
Same as above

tafaRU commented on auto_backup/data/backup_data.xml:
```diff
@@ -0,0 +1,18 @@
+            <field eval="&quot;&quot;&quot;db.backup&quot;&quot;&quot;" name="model"/>
+        </record>
+    </data>
+</openerp>
```
No newline at the end of file

tafaRU commented on auto_backup/security/ir.model.access.csv:
```diff
@@ -0,0 +1,2 @@
+id,name,model_id:id,group_id:id,perm_read,perm_write,perm_create,perm_unlink
+access_db_backup,access_db_backup,model_db_backup,,1,0,0,0
```
No newline at the end of file

tafaRU commented on auto_backup/view/bkp_conf_view.xml:
```diff
@@ -0,0 +1,108 @@
+        <menuitem parent="base.menu_config" action="action_backup_conf_form" id="backup_conf_menu"/>
+
+    </data>
+</openerp>
```
No newline at the end of file

tafaRU commented on auto_backup/model/backup_scheduler.py:
```diff
@@ -0,0 +1,428 @@
+try:
+    import pysftp
+except ImportError:
+    pass
```
Please add a debug log as written in https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#importerror

archetipo replied:
https://github.com/OCA/server-tools/pull/203#discussion-diff-37976601

tafaRU commented on auto_backup/model/backup_scheduler.py:
```diff
@@ -0,0 +1,428 @@
+    pass
+
+from openerp import models, fields, api, _
+from openerp.exceptions import except_orm, Warning
```
Use `from openerp.exceptions import Warning as UserError`, see https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#idioms as reference.

tafaRU commented on auto_backup/__openerp__.py:
```diff
@@ -0,0 +1,43 @@
+    "website": "http://www.vanroey.be/applications/bedrijfsbeheer/odoo",
+    "category": "Tools",
+    "summary": "Backups data base",
+    "depends": ['base', 'email_template'],
```
`base` could be removed since v8.0.

eLBati commented on auto_backup/model/backup_scheduler.py:
```diff
@@ -0,0 +1,425 @@
+                        _(
+                            "Couldn't backup database %s. "
+                            "Bad database administrator"
+                            "password for server running at http://%s:%s"
```
I think this message is in general wrong, because within `try` other errors than 'bad password' can happen

archetipo committed:
[FIX]  fixed, last review remarks

archetipo committed:
[FIX]  fixed, last review remarks

archetipo commented:
@eLBati  @tafaRU thanks!
all remarks should be fixed.

archetipo committed:
[FIX] travis lint check

archetipo committed:
[FIX] travis lint check

eLBati commented:
@archetipo `This branch has conflicts that must be resolved` , please rebase

dreispt commented:
Please rebase.

archetipo committed:
[FIX] backup only local db , beacause xmlrpc call of dump cause memory leak

archetipo committed:
[FIX] backup only local db , beacause xmlrpc call of dump cause memory leak

archetipo committed:
[RM] useless field

archetipo committed:
[RM] useless field

gurneyalex commented:
tests are red...

archetipo committed:
[FIX] check_dd method
[ADD] test case improved

archetipo committed:
[FIX] check_dd method
[ADD] test case improved

archetipo commented:
@gurneyalex  @eLBati  @dreispt @tafaRU  @Yenthe666 
Thanks a lot for reviews and support.

gurneyalex commented:
:+1:

eLBati commented:
tested :+1: thanks

Yenthe666 commented:
Woooh it is great to see that my initial effort is expanded and that the module got a lot cleaner.
Good job guys! Am I allowed to port over your guys changes to my Github directory again? (which in turn will update the module on the app store)

eLBati commented:
Hello @Yenthe666 you can copy the module to your repository
or you can remove it from your repo, adding a note saying "moved to https://github.com/OCA/server-tools" (as soon as this PR is merged).
Then, app store should be updated consequently

houssine78 commented on auto_backup/model/backup_scheduler.py:
```diff
@@ -0,0 +1,375 @@
+                    messageContent += _(
+                        "\nYour IP address seems to be too short.\n")
+                messageContent += _("Here is what we got instead:\n")
+        if not conn_success:
```
not sure about the variable scope. what if confs is empty? conn_success is never declared AFAICS

guewen replied:
indeed if `confs` is empty you'll get a not defined error

archetipo replied:
@houssine78  @guewen  thanks!!

houssine78 commented:
apart for my comment it's a :+1:

archetipo commented:
@houssine78 @guewen thanks, done

houssine78 commented:
@archetipo you're welcome, there is an easy way to check if you need a rebase?

archetipo commented:
@houssine78  
github check , if they commit can be merged or not , with the base branch.
int his case "This branch has no conflicts with the base branch".

houssine78 commented:
thanks @archetipo

houssine78 commented:
@dreispt @gurneyalex does this PR is waiting for something else to be merged? for what I see is has three :+1: 

thanks for your time!

yajo commented on auto_backup/data/backup_data.xml:
```diff
@@ -0,0 +1,18 @@
+            <field eval="0" name="doall"/>
+            <field eval="False" name="active"/>
+            <field eval="1" name="interval_number"/>
+            <field eval="&quot;&quot;&quot;db.backup&quot;&quot;&quot;" name="model"/>
```
All of this `"&quot;"` stuff would look much better like `"'something'"`.

yajo commented on auto_backup/__init__.py:
```diff
@@ -0,0 +1,24 @@
+##############################################################################
+
+from . import model
+from . import tests
```
No need to import tests.

yajo commented on auto_backup/README.rst:
```diff
@@ -0,0 +1,100 @@
+Known issues / Roadmap
+======================
+
+* ...
```
Remove empty sections please.

yajo commented on auto_backup/README.rst:
```diff
@@ -0,0 +1,100 @@
+Connect with an FTP Server
+--------------------------
+
+#### Keep your data safe, through an SSH tunnel!
```
Don't mix markdown with RST. See docs: http://docutils.sourceforge.net/docs/ref/rst/restructuredtext.html#sections

yajo commented on auto_backup/__openerp__.py:
```diff
@@ -0,0 +1,43 @@
+    "category": "Tools",
+    "summary": "Backups data base",
+    "depends": ['email_template'],
+    "demo": [],
```
Remove empty keys please. And [add the `external_dependencies` key](https://github.com/OCA/maintainer-tools/blob/master/template/module/__openerp__.py#L14-L17).

yajo commented on auto_backup/i18n/auto_backup.pot:
```diff
@@ -0,0 +1,141 @@
+# Translation of OpenERP Server.
```
.pot files are not needed nowadays.

yajo commented on auto_backup/model/__init__.py:
```diff
@@ -0,0 +1,23 @@
+#    You should have received a copy of the GNU General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
```
Not really a requirement, but if you use the [new-style short license headers](https://github.com/OCA/maintainer-tools/blob/master/template/module/__openerp__.py#L1-L3), you will make reviewers eyes bleed less. :smile:

yajo commented on auto_backup/__init__.py:
```diff
@@ -0,0 +1,24 @@
+#
+##############################################################################
+
+from . import model
```
[According to guidelines, this should be `models`.](https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#directories)

yajo commented on auto_backup/model/backup_scheduler.py:
```diff
@@ -0,0 +1,376 @@
+try:
+    import pysftp
+except ImportError:
+    _logger.debug('Can not import pysftp')
```
Cannot. Use WARNING level please.

yajo commented on auto_backup/model/backup_scheduler.py:
```diff
@@ -0,0 +1,376 @@
+
+    name = fields.Char(
+        string='Database', size=100, required=True,
+        default=_get_db_name,
```
[Default functions should be declared with a lambda call on self](https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#field)

yajo commented on auto_backup/model/backup_scheduler.py:
```diff
@@ -0,0 +1,376 @@
+
+    bkp_dir = fields.Char(
+        string='Backup Directory', size=100,
+        default='/odoo/backups',
```
I'd prefer a better default that ensures more directory writeability, like `~/backups`.

yajo commented on auto_backup/model/backup_scheduler.py:
```diff
@@ -0,0 +1,376 @@
+            _check_db_exist,
+            _('Error ,No such database exists'), ['name']
+        )
+    ]
```
New-api constraints should use the `@constrains` decorator unless they are SQL constraints. Could you please update this?

yajo commented on auto_backup/model/backup_scheduler.py:
```diff
@@ -0,0 +1,376 @@
+
+    @api.multi
+    def test_sftp_connection(self):
+        confs = self.search([])
```
Why this? You already are in an `@api.multi`-decorated method. Better iterate over `self` instead.

yajo commented on auto_backup/model/backup_scheduler.py:
```diff
@@ -0,0 +1,376 @@
+                        "\nYour IP address seems to be too short.\n")
+                messageContent += _("Here is what we got instead:\n")
+        if not conn_success:
+            raise except_orm(
```
That will soon be deprecated. Use a `ValidationError` instead please.

yajo commented on auto_backup/model/backup_scheduler.py:
```diff
@@ -0,0 +1,376 @@
+                    # Move to the correct directory on external server. If the
+                    # user made a typo in his path with multiple slashes
+                    # (/odoo//backups/) it will be fixed by this regex.
+                    pathToWriteTo = re.sub('/+', '/', pathToWriteTo)
```
These camelCase variables are quite anti-PEP8. Not a blocker though.

yajo commented on auto_backup/model/backup_scheduler.py:
```diff
@@ -0,0 +1,376 @@
+                    # Close the SFTP session.
+                    srv.close()
+                except Exception as e:
+                    _logger.debug(
```
Better `_logger.exception` IMHO.

yajo commented on auto_backup/model/backup_scheduler.py:
```diff
@@ -0,0 +1,376 @@
+                try:
+                    self.remove_folder(rec)
+                except Exception as e:
+                    _logger.debug(
```
Better `_logger.exception` IMHO.

yajo commented on auto_backup/model/backup_scheduler.py:
```diff
@@ -0,0 +1,376 @@
+
+        return True
+
+    def send_notification(self, rec, e):
```
`@api.model`

yajo commented on auto_backup/model/backup_scheduler.py:
```diff
@@ -0,0 +1,376 @@
+            ir_mail_server.send_email(msg)
+
+        except Exception as e:
+            _logger.debug(
```
Again Better `_logger.exception` IMHO.

yajo commented on auto_backup/model/backup_scheduler.py:
```diff
@@ -0,0 +1,376 @@
+    # to keep back-ups locally and only wants them on the SFTP
+    # (NAS for example) there wouldn't be any writing to the
+    # remote server if this if statement was before the SFTP write method
+    # right above this comment.
```
Use Python docstrings please. (`__doc__`)

yajo commented on auto_backup/model/backup_scheduler.py:
```diff
@@ -0,0 +1,376 @@
+                delta = now - createtime
+                if delta.days >= rec.daystokeep:
+                    _logger.debug(
+                        'Remove local file...'
```
INFO, please. And add the path of the removed file to the logs.

yajo commented on auto_backup/view/bkp_conf_view.xml:
```diff
@@ -0,0 +1,100 @@
+        <record model="ir.ui.view" id="view_backup_conf_form">
+            <field name="name">Configure Backup</field>
+            <field name="model">db.backup</field>
+            <field name="type">form</field>
```
remove this line

yajo commented on auto_backup/view/bkp_conf_view.xml:
```diff
@@ -0,0 +1,100 @@
+            <field name="model">db.backup</field>
+            <field name="type">form</field>
+            <field name="arch" type="xml">
+                <form string="Test">
```
test?

yajo commented on auto_backup/view/bkp_conf_view.xml:
```diff
@@ -0,0 +1,100 @@
+            <field name="arch" type="xml">
+                <form string="Test">
+                    <group col="4" colspan="4">
+                        <separator col="2" string="Local backup configuration"/>
```
You can just do `<group string="blabla">` in the corresponding group below.

yajo commented on auto_backup/view/bkp_conf_view.xml:
```diff
@@ -0,0 +1,100 @@
+                        <field name="daystokeep" attrs="{'invisible': [('autoremove','=',False)]}"/>
+                    </group>
+                    <group col="4" colspan="4">
+                        <separator col="2" string="SFTP"/>
```
Again an unneeded separator IMHO.

yajo commented on auto_backup/view/bkp_conf_view.xml:
```diff
@@ -0,0 +1,100 @@
+                        <b>Warning:</b>
+                        Use SFTP with caution! This writes files to external servers under the path you specify.
+                    </div>
+                    <group>
```
Add `attrs="{'invisible': [('sftpwrite','=',False)]}"` to the group instead of adding it to every field.

yajo commented on auto_backup/model/backup_scheduler.py:
```diff
@@ -0,0 +1,376 @@
+                os.makedirs(rec.bkp_dir)
+            # Create name for dumpfile.
+            bkp_file = '%s_%s.dump.zip' % (
+                time.strftime('%d_%m_%Y_%H_%M_%S'),
```
Filename should be `%Y_%m_%d_%H_%M_%S` instead, so databases get automatically ordered by name in evey file manager.

Yenthe666 replied:
:+1: good catch!

yajo commented on auto_backup/model/backup_scheduler.py:
```diff
@@ -0,0 +1,376 @@
+            # Create name for dumpfile.
+            bkp_file = '%s_%s.dump.zip' % (
+                time.strftime('%d_%m_%Y_%H_%M_%S'),
+                rec.name)
```
IMHO we should not rely on the database name for the generated file name. As I said before, **you should not be able to choose a database**, so instead I think it's better if the sysadmin just creates a folder for each database where all files are called the same.

I mean, instead of saving in `~/backup/2016_02_29_15_31_21_mydb.dump.zip`, force the sysadmin to save them in `~/backup/mydb/2016_02_29_15_31_21.dump.zip`, so he can use the directory tree structure he wants and all files are named equally. Better for scripting etc.

yajo commented on auto_backup/model/backup_scheduler.py:
```diff
@@ -0,0 +1,376 @@
+                        port=portHost)
+                    # Move to the correct directory on external server. If the
+                    # user made a typo in his path with multiple slashes
+                    # (/odoo//backups/) it will be fixed by this regex.
```
IMHO if the configuration is wrong, it's better to raise an exception than to auto fix it. It will prevent misconfigurations. You can add this check to an `@onchange` so that the user gets warned before bad things happen.

yajo commented on auto_backup/model/backup_scheduler.py:
```diff
@@ -0,0 +1,376 @@
+                                    ) % currentDir
+                                )
+                                # Make directory and then navigate into it
+                                srv.mkdir(currentDir, mode=777)
```
Why would a sysadmin want that my database dumps are readable by everyone by default? I already own the folder, so use a stricter mode please.

yajo commented on auto_backup/static/description/no_index.html:
```diff
@@ -0,0 +1,100 @@
+		You can even specify how long local backups and external backups should be kept, automatically!
+            </p>
+            <div class="oe_centeralign oe_websiteonly">
+                <a href="http://www.openerp.com/start?app=mail" class="oe_button oe_big oe_tacky">Start your <span class="oe_emph">free</span> trial</a>
```
<strike>I'd prefer a RST readme. Simpler to maintain. This link for example does not seem very useful (and is `odoo.com` nowadays). Not a blocker IMO though.</strike>

You have a RST readme. Why a HTML one?

yajo commented:
@archetipo Code review finished, many remarks, could you review & update please? And there are merge conflicts too.

OTOH overall this is a great addon and it's very needed indeed. Thanks for your contribution!

yajo commented on auto_backup/data/backup_data.xml:
```diff
@@ -0,0 +1,18 @@
+            <field name="user_id" ref="base.user_root"/>
+            <field eval="&quot;&quot;&quot;Backup scheduler&quot;&quot;&quot;" name="name"/>
+            <field eval="-1" name="numbercall"/>
+            <field eval="&quot;&quot;&quot;2015-07-07 11:35:28&quot;&quot;&quot;" name="nextcall"/>
```
I don't think this date should be hardcoded. Does this work actually?

yajo commented:
Hi @archetipo, I'm going to open a PR to your branch ASAP fixing those things.

yajo commented on auto_backup/__openerp__.py:
```diff
@@ -0,0 +1,43 @@
+        "VanRoey.be - Yenthe Van Ginneken, Agile Business Group,"
+        " Odoo Community Association (OCA)"
+    ),
+    'license': "AGPL-3",
```
The headers in source files are GPL-3+

Commit:
[auto_backup] Refactor.

- Follow template README.
- Remove HTML README.
- Move models to models folder.
- Model and view file names follow guidelines.
- Unused methods cleanup.
- Remove unneeded `.pot` file.
- Fix permissons.
- Follow PEP8 in names everywhere.
- Set more descriptive field names.
- Disable backups for other databases, for security.
- Remove db name from generated file, for easier cleanup.
- EAFP logic everywhere.
- More descriptive name.
- Data files moved to YAML, with cleaner ir.cron record creation.
- Add permissions for db.backup model.
- Icons.
- Update tests with new format.
- Storage method is a selectable, for easier extensibility.
- Instead of custom mailing, it just has a mail thread where you can subscribe.
- Should fix almost all comments in https://github.com/OCA/server-tools/pull/203.

Commit:
[auto_backup] Refactor.

- Follow template README.
- Remove HTML README.
- Move models to models folder.
- Model and view file names follow guidelines.
- Unused methods cleanup.
- Remove unneeded `.pot` file.
- Fix permissons.
- Follow PEP8 in names everywhere.
- Set more descriptive field names.
- Disable backups for other databases, for security.
- Remove db name from generated file, for easier cleanup.
- EAFP logic everywhere.
- More descriptive name.
- Data files moved to YAML, with cleaner ir.cron record creation.
- Add permissions for db.backup model.
- Icons.
- Update tests with new format.
- Storage method is a selectable, for easier extensibility.
- Instead of custom mailing, it just has a mail thread where you can subscribe.
- Should fix almost all comments in https://github.com/OCA/server-tools/pull/203.

yajo commented:
Please @archetipo merge https://github.com/archetipo/server-tools/pull/2 to fix all these comments and merge conflict.

archetipo committed:
Merge pull request #2 from grupoesoc/PR-203-autobackup

[auto_backup] Refactor

yajo commented on auto_backup/__openerp__.py:
```diff
@@ -0,0 +1,47 @@
+#    Copyright (C) 2015 Agile Business Group <http://www.agilebg.com>
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU General Public License as published by
+#    the Free Software Foundation, either version 3 of the License, or
```
This is something I did not want to change for you, but still you say that it is licensed under AGPL, but this header points to GPL.

Please use AGPL headers (the [reduced OCA style](https://github.com/OCA/maintainer-tools/blob/master/template/module/__openerp__.py#L1-L3) ones if possible, please).

Also you need a merge to solve conflicts.

archetipo committed:
Merge branch 'porting_auto_backup_oca' of github.com:archetipo/server-tools into porting_auto_backup_oca

archetipo commented:
@Yajo  thanks

Commit:
Fix travis.

Commit:
Reduce headers.

This respects the upstream license choice (GPL/AGPL) but reduces
verbosity.

It would be ideal to have everything under AGPL though.

Commit:
Fix view format.

Commit:
Add shortcut to execute backups from the "More" menu.

Commit:
Avoid duplicated backups.

Commit:
Make sure you don't backup inside the filestore folder.

The filestore is saved in the backup, so if you save the backup in the
filestore, you'd end up with a huge backup that includes itself and the
universe may collapse.

Commit:
[FIX] This was removing all databases.

archetipo committed:
Merge pull request #3 from grupoesoc/PR-203-autobackup

[auto_backup] Fixes

archetipo committed:
FIX License type

eLBati commented:
@archetipo anything else to be done?
Can this be merged?

archetipo commented:
@eLBati  thnaks, no, nothing to do, for me it's ok

yajo commented:
:+1:
